Extension/lib/content-script/websocket.js

```diff
@@ -0,0 +1,308 @@
+
+    };
+
+    // Safari doesn't have EventTarget
+    var EventTarget = window.EventTarget || Element;
+    WebSocket.prototype = Object.create(EventTarget.prototype, {
+        CONNECTING: {value: 0},
+        OPEN: {value: 1},
+        CLOSING: {value: 2},
+        CLOSED: {value: 3},
+        addEventListener: {
+            enumerable: true,
+            value: function (ev, cb, fl) {
+                if (cb instanceof Function === false) {
+                    return;
+                }
+                var wrapped = toWrapped.get(this);
+                if (!wrapped) {
+                    return;
+                }
+                var cbb = cb.bind(this);
+                if (wrapped instanceof Wrapped) {
+                    wrapped.addEventListener(ev, cbb, fl);
+                } else {
+                    wrapped.listeners.push({ev: ev, cb: cbb, fl: fl});
+                }
+            },
+            writable: true
+        },
+        close: {
+            enumerable: true,
+            value: function (code, reason) {
+                'native';
+                var wrapped = toWrapped.get(this);
+                if (wrapped instanceof Wrapped) {
+                    wrapped.close(code, reason);
+                }
+            },
+            writable: true
+        },
+        removeEventListener: {
+            enumerable: true,
+            value: function (ev, cb, fl) {
+            },
+            writable: true
+        },
+        send: {
+            enumerable: true,
+            value: function (data) {
+                'native';
+                var wrapped = toWrapped.get(this);
+                if (wrapped instanceof Wrapped) {
+                    wrapped.send(data);
+                }
+            },
+            writable: true
+        }
+    });
+
+    WebSocket.CONNECTING = 0;
+    WebSocket.OPEN = 1;
+    WebSocket.CLOSING = 2;
+    WebSocket.CLOSED = 3;
+
+    window.WebSocket = WebSocket;
+};
+
+/**
+ * Listener for websocket wrapper messages.
+ *
+ * @param event
+ */
+function pageMessageListener(event) {
+    if (!(event.source == window &&
+        event.data.direction &&
+        event.data.direction == "from-page-script" &&
+        event.data.elementUrl &&
+        event.data.documentUrl)) {
+        return;
+    }
+
+    var message = {
+        type: 'processShouldCollapse',
+        elementUrl: event.data.elementUrl,
+        documentUrl: event.data.documentUrl,
+        requestType: "SUBDOCUMENT",
+        requestId: 0
+    };
+
+    contentPage.sendMessage(message, function (response) {
+        if (!response) {
+            return;
+        }
+
+        event.source.postMessage({
+            direction: 'to-page-script',
+            elementUrl: event.data.elementUrl,
+            documentUrl: event.data.documentUrl,
+            collapse: response.collapse
+        }, event.origin);
+    });
+}
+
+window.addEventListener("message", pageMessageListener, false);
```

Comment:
1. Move the following constants to the very beginning of the script:

"message"
"to-page-script"
"from-page-script"

2. Rename all three to something **specific** to Adguard.
Something like:
"adguard@page-script-message"
"adguard@to-page-script"
"adguard@from-page-script"